src/main/kotlin/no/nav/familie/tilbake/service/dokumentbestilling/vedtak/VedtaksbrevFritekstValidator.kt

```diff
@@ -0,0 +1,126 @@
+                                       vedtaksbrevsoppsummering: Vedtaksbrevsoppsummering,
+                                       vedtaksbrevstype: Vedtaksbrevstype) {
+        vilkårsvurdering?.let {
+            validerFritekstISærligGrunnerAnnetAvsnitt(vilkårsvurdering,
+                                                      vedtaksbrevFritekstPerioder)
+        }
+
+        if (ORDINÆR == vedtaksbrevstype) {
+            validerFritekstIFaktaAvsnitt(faktaFeilutbetaling, vedtaksbrevFritekstPerioder, avsnittMedPerioder)
+        }
+        validerOppsummeringsfritekstLengde(behandling, vedtaksbrevsoppsummering, vedtaksbrevstype)
+        validerNårOppsummeringsfritekstErPåkrevd(behandling, vedtaksbrevsoppsummering)
+    }
+
+    private fun validerNårOppsummeringsfritekstErPåkrevd(behandling: Behandling,
+                                                         vedtaksbrevsoppsummering: Vedtaksbrevsoppsummering) {
+        val revurderingIkkeOpprettetEtterKlage = behandling.årsaker.none {
+            it.type in setOf(Behandlingsårsakstype.REVURDERING_KLAGE_KA,
+                             Behandlingsårsakstype.REVURDERING_KLAGE_NFP)
+        }
+        if (Behandlingstype.REVURDERING_TILBAKEKREVING == behandling.type &&
+            revurderingIkkeOpprettetEtterKlage &&
+            vedtaksbrevsoppsummering.oppsummeringFritekst.isNullOrEmpty()) {
+            throw Feil(message = "oppsummering fritekst påkrevet for revurdering ${behandling.id}",
+                       frontendFeilmelding = "oppsummering fritekst påkrevet for revurdering ${behandling.id}",
+                       httpStatus = HttpStatus.BAD_REQUEST)
+        }
+    }
+
+    private fun validerOppsummeringsfritekstLengde(behandling: Behandling,
+                                                   vedtaksbrevsoppsummering: Vedtaksbrevsoppsummering,
+                                                   vedtaksbrevstype: Vedtaksbrevstype) {
+        val maksTekstLengde = when (vedtaksbrevstype) {
+            ORDINÆR -> 4000
+            else -> 10000
+        }
+        if (vedtaksbrevsoppsummering.oppsummeringFritekst != null &&
+            vedtaksbrevsoppsummering.oppsummeringFritekst.length >= maksTekstLengde) {
+            throw Feil(message = "Oppsummeringstekst er for lang for behandling ${behandling.id}",
+                       frontendFeilmelding = "Oppsummeringstekst er for lang for behandling ${behandling.id}",
+                       httpStatus = HttpStatus.BAD_REQUEST)
+        }
+    }
+
+    private fun validerFritekstIFaktaAvsnitt(faktaFeilutbetaling: FaktaFeilutbetaling,
+                                             vedtaksbrevFritekstPerioder: List<Vedtaksbrevsperiode>,
+                                             avsnittMedPerioder: List<PeriodeMedTekstDto>) {
+        faktaFeilutbetaling.perioder.filter { Hendelsesundertype.ANNET_FRITEKST == it.hendelsesundertype }
+                .forEach { faktaFeilutbetalingsperiode ->
```

Comment:
faktaFeilutbetalingsperiode burde vært it

Reply:
Der brukte jeg ikke it fordi det finnes en annen loop innenfor